### PR TITLE
Hotfix - Fix external stylesheets security exception

### PR DESCRIPTION
**Pull request by Naatan**
Without this you run into a security exception because cssRules is not accessible on remote stylesheets (ie. cdn's like googlefonts).

### CODE REVIEW

File: src/initStyles.js [hunk 1]
```diff
@@ -2,7 +2,9 @@ export function initStyles(shadowDom, styleTargets) {
   const useGlobal = styleTargets.includes('global')
 
   const documentSheets = useGlobal
-    ? [...document.styleSheets]
+    ? [...document.styleSheets].filter(({href: styleHref}) =>
+        !styleHref || styleHref.startsWith(`${window.location.protocol}//${window.location.host}`)
```
Comment (markmead) [inline src/initStyles.js]: Be good to see if this works - `styleHref.includes(window.location.host)`. Just a cleaner approach.

**Review by markmead — COMMENTED** [review]
Can't say I've run into this issue but will try and replicate.

**Comment by markmead** [comment]
I've replicated the issue, thanks for finding that!

We need to make sure we're running the filter on all the stylesheets, therefore we should use something like this:

```js
  const documentSheets = useGlobal
    ? [...document.styleSheets]
    : [...document.styleSheets].filter(({ title: styleTitle }) =>
        styleTargets.includes(styleTitle)
      )

  const filteredDocumentSheets = documentSheets.filter(({ href: styleHref }) =>
    styleHref?.includes(window.location.host)
  )

  const documentStyles = filteredDocumentSheets.flatMap(({ cssRules }) => [
    ...cssRules,
  ])
```

---

Be good to also bump the `version` in `package.json` to 1.2.9.

**Comment by Naatan** [comment]
Please feel free to "fork" this PR. I just pushed a quick fix that I ran into but have since moved on from this and don't have time to iterate at the moment, I barely have time to work on my own project, sorry!

**Pull request merged**